File: 🟢 Easy/349 - Intersection of Two Arrays/_349IntersectionofTwoArrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> num1 = new HashSet<>();
        Set<Integer> num2 = new HashSet<>();

        for (int n1 : nums1) {
            num1.add(n1);
        }

        for (int n2 : nums2) {
            if (num1.contains(n2)) {
                num2.add(n2);
            }
        }

        int[] res = new int[num2.size()];
        int k = 0;

        for (int n : num2) {
            res[k++] = n;
        }

        return res;
    }
}

File: 🟢 Easy/349 - Intersection of Two Arrays/_349IntersectionofTwoArrays.js
/**
 * @param {number[]} nums1
 * @param {number[]} nums2
 * @return {number[]}
 */
var intersection = function (nums1, nums2) {
    const num1 = new Set(nums1);
    const num2 = new Set(nums2);

    let res = [];
    for (let n of num1) {
        if (num2.has(n)) {
            res.push(n);
        }
    }
    return res;
};